test/faktory_worker/job/job_integration_test.exs

```diff
@@ -0,0 +1,40 @@
+defmodule FaktoryWorker.JobIntegrationTest do
+  use ExUnit.Case
+
+  alias FaktoryWorker.Job
+
+  describe "perform_async/3" do
+    test "should send a new job to faktory" do
+      faktory_name = :"Test_#{random_string()}"
+      start_supervised!({FaktoryWorker, name: faktory_name, pool: [size: 1]})
+
+      queue_name = random_string()
+
+      opts = [
+        queue: queue_name,
+        faktory_name: faktory_name
+      ]
+
+      job = Job.create_job(SomeWorker, %{hey: "there!"}, opts)
+
+      Job.perform_async(job, opts)
+
+      assert_queue_size(queue_name, 1)
```

Comment:
I'm not massively thrilled with only asserting something is in the queue here. Once we support the `FETCH` command we will be able to update this to fetch the job and assert what was sent into faktory.